controllers/topolvm_controller.go

```diff
@@ -84,37 +81,12 @@ func (c topolvmController) ensureCreated(r *LVMClusterReconciler, ctx context.Co
 	return nil
 }
 
+// ensureDeleted is a noop. Deletion will be handled by ownerref
 func (c topolvmController) ensureDeleted(r *LVMClusterReconciler, ctx context.Context, _ *lvmv1alpha1.LVMCluster) error {
-	logger := log.FromContext(ctx).WithValues("resourceManager", c.getName())
-	existingDeployment := &appsv1.Deployment{}
-
-	err := r.Client.Get(ctx, types.NamespacedName{Name: TopolvmControllerDeploymentName, Namespace: r.Namespace}, existingDeployment)
-	// already deleted in previous reconcile
-	if k8serror.IsNotFound(err) {
-		logger.Info("csi controller deleted", "TopolvmController", existingDeployment.Name)
-		return nil
-	}
-
-	if err != nil {
-		return fmt.Errorf("failed to retrieve csi controller deployment %s: %w", existingDeployment.GetName(), err)
-	}
-
-	// if not deleted, initiate deletion
-	if !existingDeployment.GetDeletionTimestamp().IsZero() {
-		// set deletion in-progress for next reconcile to confirm deletion
-		return fmt.Errorf("topolvm controller deployment %s is already marked for deletion", existingDeployment.Name)
-	}
-
-	if err = r.Client.Delete(ctx, existingDeployment); err != nil {
-		return fmt.Errorf("failed to delete topolvm controller deployment %s: %w", existingDeployment.GetName(), err)
-	}
-
-	logger.Info("initiated topolvm controller deployment deletion", "TopolvmController", existingDeployment.Name)
 	return nil
 }
 
-func (c topolvmController) setTopolvmControllerDesiredState(existing, desired *appsv1.Deployment) error {
-
+func (c topolvmController) mutateDeployment(existing, desired *appsv1.Deployment) error {
```

Comment:
We can remove this function and move its content to the inline update function. This will make it consistent with the rest.

Reply:
removed and inlined